src/index.ts

```diff
@@ -123,39 +107,35 @@ function filterFindingsToDiffScope(
   findings: ScannerFinding[],
   filePathToChangedLines: Map<string, Set<number>>,
   inputs: PluginInputs,
-  scannerFlags: ScannerFlags
+  reporter: Reporter
 ) {
   console.log(
     "Filtering the findings to just the lines which are part of the pull request..."
   );
-  const filePathToComments = new Map<string, GithubComment[]>();
   let hasHaltingError = false;
 
   for (let finding of findings) {
     const filePath = finding.fileName.replace(process.cwd() + "/", "");
     const relevantLines =
       filePathToChangedLines.get(filePath) || new Set<number>();
     for (let violation of finding.violations) {
-      if (!isInChangedLines(violation, relevantLines) && !scannerFlags.target) {
+      const isNotChangedLines = !isInChangedLines(violation, relevantLines);
+      const isEmptyTarget = !inputs.target;
+      if (isNotChangedLines && isEmptyTarget) {
```

Comment:
This resolves the "comment on all findings" issue mentioned above correct?

This is a bit of a nit so bare with me, but the booleans here require some mental energy to understand. Perhaps we can make this a little clearer to the reader:

```TS
const reportOnTarget = inputs.target && !pullRequest; // ignore target if provided unnecessarily on pull request 
const inScopeOfPr = pullRequest && isInChangedLines(violation, relevantLines);

if (!reportOnAllFiles && !inScopeOfPr) {
  continue;
}
```